packages/core/src/components/forms/inputGroup.tsx

```diff
@@ -156,6 +164,12 @@ export class InputGroup extends AbstractPureComponent<InputGroupProps, InputGrou
         }
     }
 
+    private handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const value = event.currentTarget.value;
+        this.props.onChange?.(event);
```

Comment:
I wasn't sure if it'd be reasonable to ban native `onChange` usage, let alone make such a breaking change.

Reply:
No, we won't ban it, and we can't make a breaking change like that right now. We can just update the docs and maybe create a lint rule.

Reply:
Event pooling is going away in React 17, so I think this will become less of a problem: https://legacy.reactjs.org/blog/2020/08/10/react-v17-rc.html#no-event-pooling

I added a note in the JSDoc for the new prop about why users might prefer this handler instead of `onChange`.

Reply:
sounds good.
imo the `onValueChange` signature is also an ergonomic improvement, so i anticipate it'll still have usage